app/src/main/java/org/simple/clinic/drugs/EditMedicinesEffectHandler.kt

```diff
@@ -20,10 +29,29 @@ class EditMedicinesEffectHandler @AssistedInject constructor(
   fun build(): ObservableTransformer<EditMedicinesEffect, EditMedicinesEvent> {
     return RxMobius
         .subtypeEffectHandler<EditMedicinesEffect, EditMedicinesEvent>()
+        .addTransformer(FetchPrescribedAndProtocolDrugs::class.java, fetchDrugsList(schedulersProvider.io()))
         .addConsumer(ShowNewPrescriptionEntrySheet::class.java, { uiActions.showNewPrescriptionEntrySheet(it.patientUuid) }, schedulersProvider.ui())
         .addConsumer(OpenDosagePickerSheet::class.java, { uiActions.showDosageSelectionSheet(it.drugName, it.patientUuid, it.prescribedDrugUuid) }, schedulersProvider.ui())
         .addConsumer(ShowUpdateCustomPrescriptionSheet::class.java, { uiActions.showUpdateCustomPrescriptionSheet(it.prescribedDrug) }, schedulersProvider.ui())
         .addAction(GoBackToPatientSummary::class.java, uiActions::goBackToPatientSummary, schedulersProvider.ui())
         .build()
   }
+
+  private fun fetchDrugsList(io: Scheduler): ObservableTransformer<FetchPrescribedAndProtocolDrugs, EditMedicinesEvent> {
+    return ObservableTransformer { effects ->
+
+      val protocolDrugsStream = effects
+          .observeOn(io)
+          .flatMap { userSession.requireLoggedInUser() }
+          .switchMap { facilityRepository.currentFacility(it) }
+          .switchMap { protocolRepository.drugsForProtocolOrDefault(it.protocolUuid) }
```

Comment:
Since protocols are synced once a day, it is very unlikely that they would change while on this screen. Should we convert this to a non-reactive call and simplify the overall chain?

Reply:
This is technically not a part of this PR, so this could be something we also add to Pivotal as a task and pick up later.

Reply:
I can create a non-reactive method in `ProtocolRepository`. 
Will do this.